File: app/src/main/java/com/glsebastiany/bakingapp/view/recipe/list/RecipeStepsListAdapter.java
package com.glsebastiany.bakingapp.view.recipe.list;

import com.glsebastiany.bakingapp.R;
import com.glsebastiany.bakingapp.repository.model.Step;
import com.glsebastiany.bakingapp.view.recyclerview.RVEventHandler;
import com.glsebastiany.bakingapp.view.recyclerview.RVListBaseAdapter;

class RecipeStepsListAdapter extends RVListBaseAdapter<Step> {

    private final RVEventHandler<Step> rvEventHandler;

    public RecipeStepsListAdapter(RVEventHandler<Step> rvEventHandler) {
        this.rvEventHandler = rvEventHandler;
    }

    @Override
    public int getLayoutIdForPosition(int position) {
        return R.layout.list_item_step;
    }

    @Override
    public Object getEventHandlerObjForPosition(int position) {
        return rvEventHandler;
    }
}

File: app/src/main/java/com/glsebastiany/bakingapp/view/recipe/list/RecipeStepsListAdapter.kt
package com.glsebastiany.bakingapp.view.recipe.list

import com.glsebastiany.bakingapp.R
import com.glsebastiany.bakingapp.repository.model.Step
import com.glsebastiany.bakingapp.view.recyclerview.RVEventHandler
import com.glsebastiany.bakingapp.view.recyclerview.RVListBaseAdapter

class RecipeStepsListAdapter(private val rvEventHandler: RVEventHandler<Step>) : RVListBaseAdapter<Step>() {
    override fun getLayoutIdForPosition(position: Int): Int = R.layout.list_item_step

    override fun getEventHandlerObjForPosition(position: Int): Any? = rvEventHandler
}